### DISCUSSION TITLE
When can $q become unavailable?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#11408] [General - Components / Directives / etc] [answered]
In an `api.ts` file that I `import` in several components, I have the following code:

```
import { useQuasar } from 'quasar'

export const getToken = (): string => {
  const $q = useQuasar()
  console.log($q)
  let infinoteToken = $q.localStorage.getItem('infinote-token') as string | null
(...)
```

The first few times when that code is called, I see on the console that `$q` is correct (`{version: '2.1.0', ...`). And then, from the third call on, `$q` becomes `unavailable`.

What could be the reason for such behaviour?

**Comment by pdanpdan** [comment, marked as answer]
It is a pure luck you ever get something in $q.
useQuasar should only be used in a vue component context.
Import directly {LocalStorage} from quasar when you are outside vue components.

**Reply by wsw70** [reply]
Thanks. I was refactoring my components to remove shared code from various `<script>` and centralize it, and forgot about that.
